src/apphosting/githubConnections.ts

```diff
@@ -98,39 +99,55 @@ export async function linkGitHubRepository(
   utils.logBullet(clc.bold(`${clc.yellow("===")} Import a GitHub repository`));
   // Fetch the sentinel Oauth connection first which is needed to create further GitHub connections.
   const oauthConn = await getOrCreateOauthConnection(projectId, location);
-  const existingConns = await listAppHostingConnections(projectId, location);
+  let installationId = await promptGitHubInstallation(projectId, location, oauthConn);
 
-  if (existingConns.length === 0) {
-    existingConns.push(
-      await createFullyInstalledConnection(projectId, location, generateConnectionId(), oauthConn),
+  while (installationId === ADD_ACCOUNT_CHOICE) {
+    utils.logBullet(
+      "Install the Firebase App Hosting GitHub app on a new account to enable access to those repositories",
+    );
+
+    const apphostingGitHubInstallationURL = apphostingGitHubAppInstallationURL();
+    utils.logBullet(apphostingGitHubInstallationURL);
+    await utils.openInBrowser(apphostingGitHubInstallationURL);
+    await promptOnce({
+      type: "input",
+      message:
+        "Press Enter once you have installed or configured the Firebase App Hosting GitHub app to access your GitHub repo.",
+    });
+    installationId = await promptGitHubInstallation(projectId, location, oauthConn);
+  }
+
+  let connectionMatchingInstallation = await getConnectionForInstallation(
+    projectId,
+    location,
+    installationId,
+  );
+
+  if (!connectionMatchingInstallation) {
+    connectionMatchingInstallation = await createFullyInstalledConnection(
+      projectId,
+      location,
+      generateConnectionId(),
+      oauthConn,
+      installationId,
     );
   }
 
   let repoCloneUri: string | undefined;
-  let connection: devConnect.Connection;
+
   do {
-    if (repoCloneUri === ADD_CONN_CHOICE) {
-      existingConns.push(
-        await createFullyInstalledConnection(
-          projectId,
-          location,
-          generateConnectionId(),
-          oauthConn,
-          /** withNewInstallation= */ true,
-        ),
-      );
+    if (repoCloneUri === MANAGE_INSTALLATION_CHOICE) {
+      await manageInstallation(connectionMatchingInstallation);
     }
 
-    const selection = await promptCloneUri(projectId, existingConns);
-    repoCloneUri = selection.cloneUri;
-    connection = selection.connection;
-  } while (repoCloneUri === ADD_CONN_CHOICE);
+    repoCloneUri = await promptCloneUri(projectId, connectionMatchingInstallation);
+  } while (repoCloneUri === MANAGE_INSTALLATION_CHOICE);
 
   // Ensure that the selected connection exists in the same region as the backend
```

Comment:
where's the logic that ensures this? or is this a todo?

Reply:
ah this seems like it was an old TODO. Will remove. When fetching or creating a connection we use the region that the user wants to create their backend in.